### PR TITLE
fix logging to install hook listener

### PR DESCRIPTION
**Pull request by kwin**
log as early as possible
refactor logging interfaces a bit

### CODE REVIEW

File: accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/history/InstallationLogger.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package biz.netcentric.cq.tools.actool.history;
+
+import org.slf4j.Logger;
+
+import biz.netcentric.cq.tools.actool.api.InstallationLog;
+
+// TODO: remove extends Installation log to completely separate writing from reading
+public interface InstallationLogger extends InstallationLog {
```
Comment (kwin) [inline accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/history/InstallationLogger.java]: Add description: This is for writing to logs. Only SPI.

File: accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/installhook/AcToolInstallHook.java [hunk 2]
```diff
@@ -69,17 +69,10 @@ public void execute(InstallContext context) throws PackageException {
                 }
 
                 if (!history.isSuccess()) {
-                    for (HistoryEntry entry : history.getErrors()) {
```
Comment (ghenzler) [inline accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/installhook/AcToolInstallHook.java]: Now that we properly log we don't need to re-append errors

File: accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/installhook/AcToolInstallHook.java [hunk 3]
```diff
@@ -69,17 +69,10 @@ public void execute(InstallContext context) throws PackageException {
...
-                        log(entry.toString(), context.getOptions());
-                    }
-                    throw new PackageException(
-                            "Could not install configurations. Check log for detailed error message!");
+                    throw new PackageException("AC Tool installation failed with "
+                            +history.getErrors().size()+ " errors. Check log for detailed error message(s)!");
                 } else {
-                    // convert to correct (HTML) linebreaks for the package manager
-                    String log = history.toString().replaceAll("\\\n", "<br />");
-                    log(log, context.getOptions());
```
Comment (ghenzler) [inline accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/installhook/AcToolInstallHook.java]: Removed "aggregated one line log" - this was still in in addition to now proper logging

File: accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/history/ProgressTrackerListenerInstallationLogger.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @Override
+    public void addError(String error) {
+        listener.onError(ProgressTrackerListener.Mode.TEXT, MSG_IDENTIFIER_ERROR + error, null);
+        super.addError(error);
+    }
+
+    @Override
+    public void addError(String error, Throwable t) {
```
Comment (ghenzler) [inline accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/history/ProgressTrackerListenerInstallationLogger.java]: improved this by added another method to super class

File: accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/history/ProgressTrackerListenerInstallationLogger.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    private final ProgressTrackerListener listener;
+
+    public ProgressTrackerListenerInstallationLogger(ProgressTrackerListener listener) {
+        super();
+        this.listener = listener;
+    }
+
+    @Override
+    protected void addWarning(String warning) {
+        listener.onMessage(ProgressTrackerListener.Mode.TEXT, MSG_IDENTIFIER_WARNING + warning, "");
```
Comment (ghenzler) [inline accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/history/ProgressTrackerListenerInstallationLogger.java]: The WARNING/ERROR prefixes were missing in install hook log

File: accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/history/ProgressTrackerListenerInstallationLogger.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        } else {
+            e = new PackageException("Unexpected low-level error during AC Tool installation: " + t.getMessage(), t);
+        }
+        listener.onError(ProgressTrackerListener.Mode.TEXT, error, e);
+        super.addError(error, e);
+    }
+
+    @Override
+    protected void addVerboseMessage(String message) {
+        // no logging of verbose log in installation log (verbose log can be looked up via JMX)
```
Comment (ghenzler) [inline accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/history/ProgressTrackerListenerInstallationLogger.java]: it just gets to long hiding the important information

**Pull request merged**